src/server/utilities/cspHeader/directives.js

```diff
@@ -0,0 +1,430 @@
+  if (!isLive && isAmp) return directives.frameSrc.ampNonLive.sort();
+  if (!isLive && !isAmp) return directives.frameSrc.canonicalNonLive.sort();
+  if (isLive && isAmp) return directives.frameSrc.ampLive.sort();
+  return directives.frameSrc.canonicalLive.sort();
+};
+
+export const generateImgSrc = ({ isAmp, isLive }) => {
+  if (!isLive && isAmp) return directives.imgSrc.ampNonLive.sort();
+  if (!isLive && !isAmp) return directives.imgSrc.canonicalNonLive.sort();
+  if (isLive && isAmp) return directives.imgSrc.ampLive.sort();
+  return directives.imgSrc.canonicalLive.sort();
+};
+
+export const generateScriptSrc = ({ isAmp, isLive }) => {
+  if (!isLive && isAmp) return directives.scriptSrc.ampNonLive.sort();
+  if (!isLive && !isAmp) return directives.scriptSrc.canonicalNonLive.sort();
+  if (isLive && isAmp) return directives.scriptSrc.ampLive.sort();
+  return directives.scriptSrc.canonicalLive.sort();
+};
+
+export const generateStyleSrc = ({ isAmp, isLive }) => {
+  if (!isLive && isAmp) return directives.styleSrc.ampNonLive.sort();
+  if (!isLive && !isAmp) return directives.styleSrc.canonicalNonLive.sort();
+  if (isLive && isAmp) return directives.styleSrc.ampLive.sort();
+  return directives.styleSrc.canonicalLive.sort();
+};
+
+export const generateMediaSrc = ({ isAmp, isLive }) => {
+  if (!isLive && isAmp) return directives.mediaSrc.ampNonLive.sort();
+  if (!isLive && !isAmp) return directives.mediaSrc.canonicalNonLive.sort();
+  if (isLive && isAmp) return directives.mediaSrc.ampLive.sort();
+  return directives.mediaSrc.canonicalLive.sort();
+};
+
+export const generateWorkerSrc = ({ isAmp }) =>
+  isAmp
+    ? ['blob:', '*.bbc.co.uk', '*.bbc.com']
+    : ['blob:', "'self'", '*.bbc.co.uk', '*.bbc.com'];
+
+/*
+ * On localhost these CSP headers currently only apply on the production build.
+ * `yarn build && yarn start` & visit a localhost URL.
+ * View the developer console for errors.
+ */
+export const cspDirectives = ({
+  isAmp,
+  isLive,
+  reportOnlyOnLive,
+  service,
+}) => ({
+  directives: {
+    'default-src': generateDefaultSrc(),
+    'child-src': generateChildSrc({ isAmp }),
+    'connect-src': [
+      'afrique',
+      'arabic',
+      'hindi',
+      'japanese',
+      'mundo',
+      'portuguese',
+      'urdu',
+    ].includes(service)
+      ? "'self' https:"
+      : generateConnectSrc({ isAmp, isLive }),
+    'font-src': generateFontSrc({ isAmp, isLive }),
+    'frame-src': generateFrameSrc({ isAmp, isLive }),
+    'img-src': generateImgSrc({ isAmp, isLive }),
+    'script-src': generateScriptSrc({ isAmp, isLive }),
+    'style-src': generateStyleSrc({ isAmp, isLive }),
+    'media-src':
+      service === 'mundo'
+        ? "'self' blob: https:"
+        : generateMediaSrc({ isAmp, isLive }),
+    'worker-src': generateWorkerSrc({ isAmp }),
+    'report-to': 'worldsvc',
+    'upgrade-insecure-requests': [],
+  },
+  reportOnly: reportOnlyOnLive,
```

Comment:
Should remove the `reportOnlyOnLive` parameter and just check `service === 'japanese'` here? Would save having to update that logic in 2 places.